File: src/main/java/org/arl/fjage/Performative.java
/******************************************************************************

Copyright (c) 2013, Mandar Chitre

This file is part of fjage which is released under Simplified BSD License.
See file LICENSE.txt or go to http://www.opensource.org/licenses/BSD-3-Clause
for full license details.

******************************************************************************/

package org.arl.fjage;


/**
 * An action represented by a message. The performative actions are a subset of the
 * FIPA ACL recommendations for interagent communication.
 *
 * @author  Mandar Chitre
 */
public enum Performative {

  /**
   * Request an action to be performed.
   */
  REQUEST,

  /**
   * Agree to performing the requested action.
   */
  AGREE,

  /**
   * Refuse to perform the requested action.
   */
  REFUSE,

  /**
   * Notification of failure to perform a requested or agreed action.
   */
  FAILURE,

  /**
   * Notification of an event.
   */
  INFORM,

  /**
   * Confirm that the answer to a query is true.
   */
  CONFIRM,

  /**
   * Confirm that the answer to a query is false.
   */
  DISCONFIRM,

  /**
   * Query if some statement is true or false.
   */
  QUERY_IF,

  /**
   * Notification that a message was not understood.
   */
  NOT_UNDERSTOOD,

  /**
   * Call for proposal.
   */
  CFP,

  /**
   * Response for CFP.
   */
  PROPOSE,

  /**
   * Cancel pending request.
   */
  CANCEL

}

File: src/main/python/fjage/Performative.py
class Performative:
    """An action represented by a message."""
    REQUEST             = "REQUEST"             # Request an action to be performed.
    AGREE               = "AGREE"               # Agree to performing the requested action.
    REFUSE              = "REFUSE"              # Refuse to perform the requested action.
    FAILURE             = "FAILURE"             # Notification of failure to perform a requested or agreed action.
    INFORM              = "INFORM"              # Notification of an event.
    CONFIRM             = "CONFIRM"             # Confirm that the answer to a query is true.
    DISCONFIRM          = "DISCONFIRM"          # Confirm that the answer to a query is false.
    QUERY_IF            = "QUERY_IF"            # Query if some statement is true or false.
    NOT_UNDERSTOOD      = "NOT_UNDERSTOOD"      # Notification that a message was not understood.
    CFP                 = "CFP"                 # Call for proposal.
    PROPOSE             = "PROPOSE"             # Response for CFP.
    CANCEL              = "CANCEL"              # Cancel pending request.
